docs/source/advanced/link-prediction.rst

```diff
@@ -0,0 +1,206 @@
+
+    .. math::
+        loss = -log(\dfrac{exp(pos_score)}{\sum_{1=0}^N exp(score_i)})
+
+    where ``pos_score`` is the score of the positive edge. ``score_i`` is the score of the i-th edge. In total, there are $N+1$ edges, within which there is 1 positive edge and N negative edges.
+
+Selecting the Negative Sampler
+------------------------------
+GraphStorm provides a wide list of negative samplers:
+
+* **Uniform negative sampling**: Given ``N`` training edges under edge type ``(src_t, rel_t, dst_t)`` and the number of negatives set to K, uniform negative sampling randomly samples K nodes from ``dst_t`` for each training edge. It corrupts the training edge to form K negative edges by replacing its destination node with sampled negative nodes. In total, it will sample N * K negative nodes.
+
+    * ``uniform``: Uniformly sample K negative edges for each positive edge.
+
+    * ``fast_uniform``: same as ``uniform`` except that the sampled subgraphs
+    will not exclude edges with ``val_mask`` and ``test_mask``.
+
+    * ``all_etype_uniform``: same as ``uniform``, but it ensures that each
+    training edge type appears in every mini-batch.
+
+* **Local joint negative sampling**: Local uniform negative sampling samples negative edges in the same way as uniform negative sampling except that all the negative nodes are sampled from the local graph partition.
+
+    * ``localuniform``: Uniformly sample K negative edges for each positive edge.
+    However the negative nodes are sampled from the local graph partition
+    instead of being sampled globally.
+
+    * ``fast_localuniform``: same as ``localuniform`` except that the sampled subgraphs
+    will not exclude edges with ``val_mask`` and ``test_mask``. Please see the details in :ref:`speedup_lp_training_label`.
+
+* **Joint negative sampling**: Given ``N`` training edges under edge type ``(src_t, rel_t, dst_t)`` and the number of negatives set to K, joint negative sampling randomly samples K nodes from ``dst_t`` for every K training edges. For these K training edges, it corrupts each edge to form K negative edges by replacing its destination node with the same set of negative nodes. In total, it only needs to sample $N$ negative nodes. (We suppose N is dividable by K for simplicity.)
+
+    * ``joint``: Sample K negative nodes for every K positive edges.
+    The K positive edges will share the same set of negative nodes
+
+    * ``fast_joint``: same as ``joint`` except that the sampled subgraphs
+    will not exclude edges with ``val_mask`` and ``test_mask``.
+    Please see the details in :ref:`speedup_lp_training_label`.
+
+    * ``all_etype_joint``: same as ``joint``, but it ensures that each
+    training edge type appears in every mini-batch.
+
+* **Local joint negative sampling**: Local joint negative sampling samples negative edges in the same way as joint negative sampling except that all the negative nodes are sampled from the local graph partition.
```

Comment:
duplicated to the line 107?

Reply:
107 is "Local uniform negative sampling". Updated.